### PR TITLE
Look for config file in working directory before the home directory

### PR DESCRIPTION
**Pull request by mediumTaj**
This pull request changes the order of where the SDK looks for external credentials. The SDK now looks for credentials in the project directory before looking in the home directory.

BREAKING CHANGE: SDK now looks for the config file in the working directory before looking in the
home directory

### CODE REVIEW

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs [hunk 1]
```diff
@@ -132,37 +132,37 @@ private static List<string> GetFilesToCheck()
             {
                 files.Add(userSpecifedPath);
             }
-
-            if (!string.IsNullOrEmpty(unixHomeDirectory))
+
```
Comment (lpatino10) [inline src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: Some extra whitespace here that can be removed

**Review by lpatino10 — APPROVED** [review]
Looks good minus a style comment to remove some whitespace. I'll approve anyway so that you can merge after that change if you want.

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs (hunk 1 above)
Comment (padamstx) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: @lpatino10 with the eagle eye on this one :)

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs [hunk 2]
```diff
@@ -132,37 +132,37 @@ private static List<string> GetFilesToCheck()
             {
                 files.Add(userSpecifedPath);
             }
-
-            if (!string.IsNullOrEmpty(unixHomeDirectory))
+            
+            if (!string.IsNullOrEmpty(projectDirectory))
```
Comment (padamstx) [inline src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: I'm curious why we call it "projectDirectory"...  I think technically it's the "current working directory", right?

**Review by padamstx — APPROVED** [review]
The changes look good.   I left one comment that can be ignored.
But one other question I have...  we have a function that returns a list of files (i.e. files found to exist in the search order), then we have a function that really only processes the first file in the list, then it breaks out of the for loop.   Why don't we just return a single file from the first function instead of the list (i.e. just return the first file that you find when processing the search order), then read in the contents of that one file.   That would simplify things a bit and achieve the same result I think.

**Comment by lpatino10** [comment]
I didn't even notice that but yeah I think @padamstx has a point about simplifying that logic.

**Comment by padamstx** [comment]
> I didn't even notice that but yeah I think @padamstx has a point about simplifying that logic.

Pretty sure the java core works that way too.   I'll fix that when I make these same changes there.

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs (hunk 2 above)
Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: hmm yeah thats the current working directory - for .net its different depending on the build target. Not sure what a better name could be - `projectBuildDirectory`?

Comment (padamstx) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: why not just "currentDirectory" or "cwd"?   I mean... what does "project" even mean in this context?   We're talking about the current working directory that might be in effect when the application is running, right?

**Comment by mediumTaj** [comment]
OK I'm now returning after finding the first filepath. Can you guys take another look?

**Review by padamstx — APPROVED** [review]
LGTM

**Review by lpatino10 — APPROVED** [review]

File: src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs (hunk 2 above)
Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs]: OK I changed it to currentDirectory - which makes sense from the built project's perspective.

**Pull request merged**